common/persistence/sql/sqlplugin/mysql/plugin.go

```diff
@@ -97,11 +111,67 @@ func (p *plugin) createDBConnection(cfg *config.SQL) (*sqlx.DB, error) {
 	if cfg.MaxConnLifetime > 0 {
 		db.SetConnMaxLifetime(cfg.MaxConnLifetime)
 	}
+
 	// Maps struct names in CamelCase to snake without need for db struct tags.
 	db.MapperFunc(strcase.ToSnake)
 	return db, nil
 }
 
+func registerTLSConfig(cfg *config.SQL) error {
+	host, _, err := net.SplitHostPort(cfg.ConnectAddr)
+	if err != nil {
+		return fmt.Errorf("error in host port from ConnectAddr: %v", err)
+	}
+
+	tlsConfig := &tls.Config{
```

Comment:
I would strongly suggest specifying the following parameters: 

- MinVersion for the minimum TLS version that's acceptable.  By default, the minVersion is 1.0, which pretty much should not be used. The default value for this should be 1.3. 
- CipherSuites - Ideally, this should also be set. TLS 1.3 already prevents you from choosing bad ciphersuites

I am ok if you punt on this change for this PR. But please create a github task to take these as config params

Reply:
It looks like AWS Aurora supports up to 1.2 which would be the minimum for our use case. I can hard code 1.2 if you think that is ok for other users? I think the best way would be allowing the configuration of those options via the cfg but that would take more plumbing and would need to take in account TLS for Cassandra and Kafka because I am sharing the config object.

For now I left it unchanged (with a TODO comment) and will create an issue.

Reply:
Let's tackle this separately via config. For now, I am ok with using the defaults